### ISSUE TITLE
Make strict:all and oil:basic orthogonal, and document it

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#959]
(after thinking about this some more)

@happysalada was running `osh -O strict:all` on Nix and this caused some changes like `$x` -> `${x:-}`

https://github.com/NixOS/nixpkgs/pull/127736

But I think this belongs in `oil:basic`.  Right now `strict:all` has all of these options:

- errexit
- pipefail
- nounset
- inherit_errexit
- nullglob

If the goal is to find bugs in **unmodified** bash scripts, then I think they should all be off.  Only the `strict_*` options should be on.

In other words, running shell scripts with **errexit off** is valid in OSH.  It's not recommended in Oil, which is why `oil:basic` and `oil:all` have it turned on.

----

I need to revise `doc/oil-options.md` and explain this.  Also see https://github.com/oilshell/oil/wiki/Gradually-Upgrading-Shell-to-Oil, which currently doesn't mention `strict:all`.

**Comment by andychu** [comment]
@happysalada 

After thinking about it, I think that `errexit` and so forth should still be on in `strict:all`.  I haven't properly documented what the purpose of this is, which is the real issue.

If you want to use it, you're supposed to put it at the top of your script,  as in

https://github.com/oilshell/oil/wiki/Migration-Guide

https://github.com/oilshell/oil/wiki/Gradually-Upgrading-Shell-to-Oil (I need to consolidate these pages, again a doc problem)

I think it's an error to fix bugs with `osh -O strict:all` but WITHOUT enabling those options in the script!

I don't think any documentation suggests doing that, but I think I may have suggested it on lobste.rs or something, and that idea  is erroneous.

So I'm going to change this bug to improve the docs.  Let me know if that makes sense!

**Comment by happysalada** [comment]
Thank you for the update!

In the Gradually-Upgrading-Shell-to-Oil there is no reference to strict:all ? Should there not be a step in between 1 and 2 to use strict:all before going to oil:basic?

Otherwise the docs look good !

**Comment by andychu** [comment]
I just made `strict:all` and `oil:basic` disjoint.  

In other words, I changed my mind again!  This stuff is hard to think about...  But one way to see why this is a good idea is to look at Oil's OWN usage of `strict:all`:

https://github.com/oilshell/oil/blob/master/build/actions.sh#L11

Our scripts run under both bash and Oil.  So you need to have **both** `set -o errexit` and `shopt --set strict:all` !  That is, `strict:all` doesn't do anything in bash, so you need `set -o errexit` too.

I made another pass over `doc/options.md`, and it still needs more editing.  The docs will clarify things -- if I had done some more "documentation-driven development", perhaps the waffling back and forth would have been avoided :)

Thanks again for trying Oil -- this has been really helpful!

**Comment by happysalada** [comment]
Got it thanks. The last PR I made almost made it onto master. I'm preparing some more PRs coming, I'll tag you on those.
Thank you for this!
